compiler/src/main/scala/dx/compiler/WorkflowCompiler.scala

```diff
@@ -414,7 +586,89 @@ case class WorkflowCompiler(extras: Option[Extras],
     val execTreeDetails = Map(
         "execTree" -> JsString(CodecUtils.gzipAndBase64Encode(execTree.toString))
     )
-    val details = wfMetaDetails ++ sourceDetails ++ dxLinks ++ delayDetails ++ execTreeDetails
+
+    // Collect native app dependencies
+    val nativeAppDependencies: Vector[String] =
+      Set
+        .from(
+            workflow.stages
+              .collect { stage =>
+                executableDict(stage.calleeName) match {
+                  case CompiledExecutable(
+                      Application(_,
+                                  _,
+                                  _,
+                                  _,
+                                  _,
+                                  ExecutableKindNative(_, _, _, _, _),
+                                  _,
+                                  _,
+                                  _,
+                                  _,
+                                  _,
+                                  _),
+                      dxExec,
+                      _,
+                      _
+                      ) =>
+                    dxExec.id
+                }
+              }
+              .filter(_.startsWith("app-"))
+        )
+        .toVector
+    val nativeAppDependenciesDetails = nativeAppDependencies match {
+      case nativeAppDependencies if nativeAppDependencies.length > 0 =>
+        Map(
+            Constants.NativeAppDependencies -> JsArray(
+                nativeAppDependencies.map(id => JsString(id))
+            )
+        )
+      case _ => Map.empty
+    }
+
+    // Collect file dependencies from inputs & private variables
+    val fileInputParams = workflow.inputs.map(i => i._1).filter(param => param.dxType == Type.TFile)
+    val fileDependencies =
+      Set
+        .from(
+            fileInputParams
+              .flatMap(CompilerUtils.fileDependenciesFromParam) ++ workflow.varFileDependencies
+        )
+        .toVector
+    val fileDependenciesDetails = fileDependencies match {
+      case fileDependencies if !fileDependencies.isEmpty =>
+        Map(Constants.FileDependencies -> JsArray(fileDependencies.map(s => JsString(s))))
+      case _ => Map.empty
+    }
+
+    // Collect Docker registry credentials URI
+    val dockerRegistryCredentialsUri: Option[String] = extras.flatMap(_.dockerRegistry) match {
+      case Some(DockerRegistry(_, credentials, _, _)) => Some(credentials)
+      case _                                          => None
+    }
+    val dockerRegistryCredentialsUriDetails = dockerRegistryCredentialsUri match {
```

Comment:
Here you can do `dockerRegistryCredentialsUri.map(cred => Map(Constants.DockerRegistryCredentialsUri -> JsString(cred))).getOrElse(Map.empty)`

Reply:
Cool, I didn't realize `Option.map` method exists. I guess it makes sense if you think of Option as a special case of collection containing 0-1 items. Done. @jdidion